### PR TITLE
Initialize empty producers using mobility and transmissibility ratio 

### PR DESCRIPTION
**Pull request by totto82**
The approach was suggested by @jalvestad 
And initial testing on norne and model2 is promising.

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
benchmark please

File: opm/simulators/wells/StandardWell_impl.hpp [hunk 1]
```diff
@@ -2329,6 +2333,31 @@ namespace Opm
...
+                double total_mobility = 0.0;
+                for (int component = 0; component < num_components_; ++component) {
+                    total_mobility += mob[component].value();
+                }
+                for (int component = 0; component < num_components_; ++component) {
+                    perfRates[perf * num_components_ + component] += well_tw_fraction * mob[component].value() / total_mobility;
+                }
+            }
+        }
+
```
Comment (GitPaean) [inline opm/simulators/wells/StandardWell_impl.hpp:2360]: With the above modification, will `tot_surf_rate==0` still happens in function `computeConnectionDensities`? I think with this, perfRates will not be all zero anymore? Do I miss something here?

File: opm/simulators/wells/StandardWell_impl.hpp [hunk 2]
```diff
@@ -2070,20 +2070,25 @@ namespace Opm
...
+                        case Phase::WATER:
+                            mix[FluidSystem::waterCompIdx] = 1.0;
+                            break;
+                        default:
+                            // No others supported.
+                            break;
+                        }
+                    // For the rest of the perforation without flow we use mix from the above perforation.
+                    } else {
+                        mix = x;
```
Comment (GitPaean) [inline opm/simulators/wells/StandardWell_impl.hpp:2091]: It might make no big difference and `tot_surf_rate == 0.` probably not happen very often.  Considering the flow direction, should not the perforation below be used here?   Following this line, should we do the iteration from bottom to up instead of from up to bottom?  I am not very familiar with this part of the code, so it is just some questions here.

File: opm/simulators/wells/StandardWell_impl.hpp (hunk 1 above)
Comment (totto82) [reply opm/simulators/wells/StandardWell_impl.hpp:2360]: Yes it can be zero. If for instance the bottom perforation have zero rates, but the top has a non-trivial rate. This happens often for wells where cross flow is not allowed. For instance in model 2.

File: opm/simulators/wells/StandardWell_impl.hpp (hunk 2 above)
Comment (totto82) [reply opm/simulators/wells/StandardWell_impl.hpp:2091]: See comment above. The tot_surf_rate is accumulated from the bottom. If tot_surf_rate is zero for a perforation most probably all underlying perforations have zero rate. i.e. we can not use the mixture from the bottom. The logic of using the mixture from the top is that if fluid is not entering from the reservoir the well bore will fill still be filled up with something. And that something must come from the top.

**Comment by totto82** [comment]
benchmark please

**Comment by ytelses** [comment]
Benchmark result overview: 




| Test    | Configuration                                     | Relative |
| ------- | ------------------------------------------------- | -------- |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 1        | 0.989    |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 8        | 1.024    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 1        | 0.999    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 8        | 0.997    |
| opm-git | OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 1 | 1        |
| opm-git | OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 8 | 1        |
* Speed-up = Total time master / Total time pull request. Above 1.0 is an improvement. *



View result details @ https://www.ytelses.com/opm/?page=result&id=1163

**Comment by ytelses** [comment]
Benchmark result overview: 




| Test    | Configuration                                     | Relative |
| ------- | ------------------------------------------------- | -------- |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 1        | 0.989    |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 8        | 1.024    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 1        | 0.996    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 8        | 1.003    |
| opm-git | OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 1 | 0.999    |
| opm-git | OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 8 | 0.999    |
* Speed-up = Total time master / Total time pull request. Above 1.0 is an improvement. *



View result details @ https://www.ytelses.com/opm/?page=result&id=1164

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
This gives a slightly more accurate initialization of the pressure column down the standard well. It gives some noise in the results and convergence of several of the test cases but all are minor. I will start the process of merging by updating the reference data

**Comment by totto82** [comment]
jenkins build this update_data please

**Comment by totto82** [comment]
jenkins build this update_data please

**Comment by totto82** [comment]
jenkins build this opm-tests=528 please

**Comment by totto82** [comment]
This PR does not effect the simulations much, but improves the results for some corner cases.  I will self-merge.
